### ISSUE TITLE
Duplicate `PRIResources` causing performance degradation

### ISSUE THREAD

**Issue by Youssef1313** [unoplatform/uno#11464]
### Current behavior

![image](https://user-images.githubusercontent.com/31348972/221382031-0cf886e3-7caa-4bd8-8888-11b3c2c2fc0e.png)

There is currently too many calls to `GetText` from `BuildLocalResourceDetails`. Upon investigation (by looking at a random binlog I happened to already have locally), I found that each resource file is present three times

![image](https://user-images.githubusercontent.com/31348972/221382091-b073e2a1-4aa1-41cd-9035-23bb9f3e0ad5.png)

It's best if the de-duplication is **not** done in `BuildLocalResourceDetails` itself, but avoided in build in the first place.

### Works on UWP/WinUI

None

**Comment by Youssef1313** [comment]
In addition to avoid duplicate resources, I think memoizing `GetText` might help too. @jeromelaban What do you think?

**Comment by jeromelaban** [comment]
Duplicate PRI resources in the first place is curious. Is this happening from a new app, or from uno itself?

**Comment by Youssef1313** [comment]
I haven't check a new app. Will do

**Comment by Youssef1313** [comment]
In a brand new unoapp, I found that PRIResources are not added completely, which should be fixed by #11511. Not sure why they are repeated for uno.extensions. I opened https://github.com/unoplatform/uno.extensions/issues/1224

- Should we add a warning when we find duplicate PRIResources?
- Should we silently de-duplicate? (Maybe check if WinUI already does that or not?)
- I'll also see if uno builds have the same issue.
